src/Microsoft.AspNetCore.SignalR.Client/Connection.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.IO.Pipelines;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    public class Connection : IPipelineConnection
+    {
+        private IPipelineConnection _consumerPipe;
+        private ITransport _transport;
+        private readonly ILogger _logger;
+
+        public Uri Url { get; }
+
+        public Connection(Uri url, ITransport transport, IPipelineConnection consumerPipe, ILoggerFactory _loggerFactory)
+        {
```

Comment:
validate parameters in public APIs
we should have a ctor that does not take LoggerFactory (and possibly transport - I think we should default to webSockets)